src/tests/SimTest.cpp

```diff
@@ -53,3 +79,52 @@ TEST(SimTest, Reset) {
   simulator.reset();
   ASSERT_EQ(pathfinder, simulator.getPathFinder());
 }
+
+TEST(SimTest, GetPinholeCameraRGBAObservation) {
+  SimulatorConfiguration simConfig{};
+  simConfig.scene.id = vangogh;
+
+  // do not rely on default SensorSpec default constructor to remain constant
+  auto pinholeCameraSpec = SensorSpec::create();
+  pinholeCameraSpec->sensorSubtype = "pinhole";
+  pinholeCameraSpec->sensorType = SensorType::COLOR;
+  pinholeCameraSpec->position = {0.0f, 1.5f, 5.0f};
+  pinholeCameraSpec->resolution = {100, 100};
+
+  SimulatorWithAgents simulator(simConfig);
+  AgentConfiguration agentConfig{};
+  agentConfig.sensorSpecifications = {pinholeCameraSpec};
+  simulator.addAgent(agentConfig);
+
+  Observation observation;
+  ObservationSpace obsSpace;
+  ASSERT_TRUE(simulator.getAgentObservation(
+      simConfig.defaultAgentId, pinholeCameraSpec->uuid, observation));
+  ASSERT_TRUE(simulator.getAgentObservationSpace(
+      simConfig.defaultAgentId, pinholeCameraSpec->uuid, obsSpace));
+
+  std::vector<size_t> expectedShape{
+      {static_cast<size_t>(pinholeCameraSpec->resolution[0]),
+       static_cast<size_t>(pinholeCameraSpec->resolution[1]), 4}};
+
+  ASSERT_EQ(obsSpace.spaceType, ObservationSpaceType::TENSOR);
+  ASSERT_EQ(obsSpace.dataType, esp::core::DataType::DT_UINT8);
+  ASSERT_EQ(obsSpace.shape, expectedShape);
+  ASSERT_EQ(observation.buffer->shape, expectedShape);
+
+  // Compare one pixel (center of image) with previously rendered ground truth
+  // TODO: compare more than one pixel
```

Comment:
FYI (I guess you know already), if you ever get to this TODO, Corrade's TestSuite allows you to do this easily: https://doc.magnum.graphics/magnum/classMagnum_1_1DebugTools_1_1CompareImage.html

It would mean you'd need to port this away from googletest, but looking at the other parts of this test, it seems it would benefit from that :)